tests/test_tasks.py

```diff
@@ -0,0 +1,131 @@
+    invalid_checksum = "invalid"
+    file_instance.checksum = invalid_checksum
+    file_instance.verify_checksum()
+    db.session.commit()
+    f = FileInstance.query.get(file_instance.id)
+    assert f.last_check is False
+
+    # A report must be generated for the file
+    mail = app.extensions.get("mail")
+    assert mail
+
+    recipients = "test@invenio.org"
+    sender = "test@invenio.org"
+
+    file_name = file_instance.objects[0].key
+
+    with mail.record_messages() as outbox:
+        # Configure email and validate that email was sent
+        app.config["FILES_INTEGRITY_REPORT_TEMPLATE"] = "mock_mail.html"
+        app.config["ADMIN_EMAIL_RECIPIENT"] = recipients
+        app.config["MAIL_DEFAULT_SENDER"] = sender
+
+        file_integrity_report()
+        assert len(outbox) == 1
+        mail_sent = outbox[0]
+        assert mail_sent.recipients == [recipients]
+        assert mail_sent.sender == sender
+        assert "Checksum: {}".format(invalid_checksum) in mail_sent.body
+        assert "URI: {}".format(f.uri) in mail_sent.body
+        assert "ID: {}".format(str(f.id)) in mail_sent.body
+        assert "Name: {}".format(file_name) in mail_sent.body
+
+
+def test_integrity_report_invalid_template(app, db, file_instance):
+    """Test non-existant e-mail template."""
+    assert file_instance.last_check is True
+
+    # Modify file contents to force checksum to fail
+    invalid_checksum = "invalid"
+    file_instance.checksum = invalid_checksum
+    file_instance.verify_checksum()
+    db.session.commit()
+    f = FileInstance.query.get(file_instance.id)
+    assert f.last_check is False
+
+    mail = app.extensions.get("mail")
+    assert mail
+
+    with mail.record_messages() as outbox:
+        # Remove the template, no e-mail is sent
+        app.config["FILES_INTEGRITY_REPORT_TEMPLATE"] = None
+        file_integrity_report()
+        assert len(outbox) == 0
+
+
+def test_integrity_report_invalid_addresses(app, db, file_instance):
+    """Test invalid recipient address."""
+    assert file_instance.last_check is True
+
+    # Modify file contents to force checksum to fail
+    invalid_checksum = "invalid"
+    file_instance.checksum = invalid_checksum
+    file_instance.verify_checksum()
+    db.session.commit()
+    f = FileInstance.query.get(file_instance.id)
+    assert f.last_check is False
+
+    mail = app.extensions.get("mail")
+    assert mail
+
+    with mail.record_messages() as outbox:
+        # Use mock template
+        app.config["FILES_INTEGRITY_REPORT_TEMPLATE"] = "mock_mail.html"
+
+        # Recipient is not set, mail is not sent.
+        app.config["ADMIN_EMAIL_RECIPIENT"] = None
+        file_integrity_report()
+        assert len(outbox) == 0
+
+
+def test_integrity_report_default_template(app, db, file_instance):
+    """Test invalid recipient address."""
+    assert file_instance.last_check is True
+
+    # Modify file contents to force checksum to fail
```

Comment:
maybe we can extract this in a reusable function so we improve readability here?